src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -566,6 +577,16 @@ def serialize(
                     current_event = to_check.popleft()
                     previous_event = parent_events[current_event["id"]]
 
+                # We've found the event for the quick trace so we can remove everything in the deque
+                # As they're unrelated ancestors now
+                if event_id and current_event["id"] == event_id:
```

Comment:
Just a thought for further optimization (not for this PR) but we can do additional pruning by removing paths that contain nodes that are not on the same path as the specified event, even after they've been retrieved from nodestore. This will save a few bytes and the result will have the guarantee that everything returned is relevant. Because as it is, the frontend needs to do additional pruning to be left with only the relevant path.

Reply:
And as far as I can tell, if the event id is in an orphan branch, the root branch will not be pruned right? Not big deal in this PR, but another place for additional optimization.

Reply:
So yeah, the endpoint will still return branches that aren't relevant, so there's a chance based on iteration order that we still return all n events in this response.

I definitely agree that this can be further optimization, but probably not worth the effort in the short-term IMO. We're only saving a few bytes and the runtime won't be very different as far as I'm aware (like the frontend pruning is fast already, and we'd still have to move it to the backend too)

Though, this should be the first thing we target if we're seeing performance issues because of the size of the response